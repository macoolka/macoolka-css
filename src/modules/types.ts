export type Level = 'small' | 'medium' | 'large';
