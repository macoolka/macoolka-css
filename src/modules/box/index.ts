/**
 * convert number type to px or precent
 * @getter
 */
import { UnitProps } from '../../basic/';
import { Rule } from '../../base/rule';

export type Theme = {
    box: {
        width: {
            content: number,
            gutter: number;
            nav: number;
            third: number;
        },
        size: {
            small: number,
            medium: number,
            large: number,
            xlarge: number,
        }
    }

};
export const theme: Theme = {
    box: {
        width: {
            content: 1160,
            gutter: 40,
            nav: 256,
            third: 355,
        },
        size: {
            small: 18,
            medium: 24,
            large: 36,
            xlarge: 48,
        },
    },

};
type SUnit = 0 | 1 | 2 | 4 | 6 | 8 | 12 | 16 | 24 | 36 | 48 | 64 | 124;
type SProps = {
    /**
     * The property is used to setting the margin properties in one declaration
     */
    mkMargin: SUnit,
    /**
     * The property is used to setting the top margin of an element.
     */
    mkMarginTop: SUnit,
    /**
     * The property is used to setting the bottom margin of an element.
     */
    mkMarginBottom: SUnit,
    /**
     * The property is used to setting the left margin of an element.
     */
    mkMarginLeft: SUnit,
    /**
     * The property is used to setting the right margin of an element.
     */
    mkMarginRight: SUnit,
    /**
     * The property is used to setting the x margin of an element.
     */
    mkMarginH: SUnit,
    /**
     * The property is used to setting the y margin of an element.
     */
    mkMarginV: SUnit,
    /**
     * The property is used to setting the padding of an element.
     */
    mkPadding: SUnit,
    /**
     * The property is used to setting the top padding of an element.
     */
    mkPaddingTop: SUnit,
    /**
     * The property is used to setting the bottom padding of an element.
     */
    mkPaddingBottom: SUnit,
    /**
     * The property is used to setting the left padding of an element.
     */
    mkPaddingLeft: SUnit,
    /**
     * The property is used to setting the right padding of an element.
     */
    mkPaddingRight: SUnit,
    /**
     * The property is used to setting the x padding of an element.
     */
    mkPaddingH: SUnit,
    /**
     * The property is used to setting the y padding of an element.
     */
    mkPaddingV: SUnit,

};
type EProps = {
    mkWidth: 'full' | 'fullScreen' | 'content' | 'nav' | 'gutter'
     | 'third' | 'small' | 'medium' | 'large' | 'xlarge';
    mkMinWidth: 'full' | 'fullScreen' | 'content' | 'nav' | 'gutter'
     | 'third' | 'small' | 'medium' | 'large' | 'xlarge';
    mkMaxWidth: 'full' | 'fullScreen' | 'content' | 'nav'
     | 'gutter' | 'third' | 'small' | 'medium' | 'large' | 'xlarge';
    mkHeight: 'full' | 'fullScreen' | 'small' | 'medium' | 'large' | 'xlarge';
    mkMinHeight: 'full' | 'fullScreen' | 'small' | 'medium' | 'large' | 'xlarge';
    mkMaxHeight: 'full' | 'fullScreen' | 'small' | 'medium' | 'large' | 'xlarge';

    mkSquare: 'small' | 'medium' | 'large' | 'xlarge';
};
export type Props = EProps & SProps;
export const rule: Rule<SProps, EProps, UnitProps, Theme> = {
    rule: {
        /**
         * The property is used to setting the margin properties in one declaration
         */
        mkMargin: 'margin',
        /**
         * The property is used to setting the top margin of an element.
         */
        mkMarginTop: ('marginTop'),
        /**
         * The property is used to setting the bottom margin of an element.
         */
        mkMarginBottom: ('marginBottom'),
        /**
         * The property is used to setting the left margin of an element.
         */
        mkMarginLeft: ('marginLeft'),
        /**
         * The property is used to setting the right margin of an element.
         */
        mkMarginRight: ('marginRight'),
        /**
         * The property is used to setting the x margin of an element.
         */
        mkMarginH: (['marginLeft', 'marginRight']),
        /**
         * The property is used to setting the y margin of an element.
         */
        mkMarginV: (['marginTop', 'marginBottom']),
        /**
         * The property is used to setting the padding of an element.
         */
        mkPadding: ('padding'),
        /**
         * The property is used to setting the top padding of an element.
         */
        mkPaddingTop: ('paddingTop'),
        /**
         * The property is used to setting the bottom padding of an element.
         */
        mkPaddingBottom: ('paddingBottom'),
        /**
         * The property is used to setting the left padding of an element.
         */
        mkPaddingLeft: ('paddingLeft'),
        /**
         * The property is used to setting the right padding of an element.
         */
        mkPaddingRight: ('paddingRight'),
        /**
         * The property is used to setting the x padding of an element.
         */
        mkPaddingH: ['paddingLeft', 'paddingRight'],
        /**
         * The property is used to setting the y padding of an element.
         */
        mkPaddingV: (['paddingTop', 'paddingBottom']),

    },
    ruleEnum: {
        mkWidth: {
            fullScreen: {
                width: '100vw',
            },
            full: {
                width: '100%',
            },
            content: {
                width: (t) => t.box.width.content,
            },
            nav: {
                width: (t) => t.box.width.nav,
            },
            gutter: {
                width: (t) => t.box.width.gutter,
            },
            third: {
                width: (t) => t.box.width.third,
            },
            small: {
                width: (t) => t.box.size.small,
            },
            medium: {
                width: (t) => t.box.size.medium,
            },
            large: {
                width: (t) => t.box.size.large,
            },
            xlarge: {
                width: (t) => t.box.size.xlarge,
            },
        },
        mkMinWidth: {
            fullScreen: {
                minWidth: '100vw',
            },
            full: {
                minWidth: '100%',
            },
            content: {
                minWidth: (t) => t.box.width.content,
            },
            nav: {
                minWidth: (t) => t.box.width.nav,
            },
            gutter: {
                minWidth: (t) => t.box.width.gutter,
            },
            third: {
                minWidth: (t) => t.box.width.third,
            },
            small: {
                minWidth: (t) => t.box.size.small,
            },
            medium: {
                minWidth: (t) => t.box.size.medium,
            },
            large: {
                minWidth: (t) => t.box.size.large,
            },
            xlarge: {
                minWidth: (t) => t.box.size.xlarge,
            },
        },
        mkMaxWidth: {
            fullScreen: {
                maxWidth: '100vw',
            },
            full: {
                maxWidth: '100%',
            },
            content: {
                maxWidth: (t) => t.box.width.content,
            },
            nav: {
                maxWidth: (t) => t.box.width.nav,
            },
            gutter: {
                maxWidth: (t) => t.box.width.gutter,
            },
            third: {
                maxWidth: (t) => t.box.width.third,
            },
            small: {
                maxWidth: (t) => t.box.size.small,
            },
            medium: {
                maxWidth: (t) => t.box.size.medium,
            },
            large: {
                maxWidth: (t) => t.box.size.large,
            },
            xlarge: {
                maxWidth: (t) => t.box.size.xlarge,
            },
        },
        mkHeight: {
            fullScreen: {
                height: '100vh',
            },
            full: {
                height: '100%',
            },
            small: {
                height: (t) => t.box.size.small,
            },
            medium: {
                height: (t) => t.box.size.medium,
            },
            large: {
                height: (t) => t.box.size.large,
            },
            xlarge: {
                height: (t) => t.box.size.xlarge,

            },
        },
        mkMaxHeight: {
            fullScreen: {
                maxHeight: '100vh',
            },
            full: {
                maxHeight: '100%',
            },
            small: {
                maxHeight: (t) => t.box.size.small,
            },
            medium: {
                maxHeight: (t) => t.box.size.medium,
            },
            large: {
                maxHeight: (t) => t.box.size.large,
            },
            xlarge: {
                maxHeight: (t) => t.box.size.xlarge,

            },
        },
        mkMinHeight: {
            fullScreen: {
                minHeight: '100vh',
            },
            full: {
                minHeight: '100%',
            },
            small: {
                minHeight: (t) => t.box.size.small,
            },
            medium: {
                minHeight: (t) => t.box.size.medium,
            },
            large: {
                minHeight: (t) => t.box.size.large,
            },
            xlarge: {
                minHeight: (t) => t.box.size.xlarge,

            },
        },
        mkSquare: {
            small: {
                width: (t) => t.box.size.small,
                height: (t) => t.box.size.small,
            },
            medium: {
                width: (t) => t.box.size.medium,
                height: (t) => t.box.size.medium,
            },
            large: {
                width: (t) => t.box.size.large,
                height: (t) => t.box.size.large,
            },
            xlarge: {
                width: (t) => t.box.size.xlarge,
                height: (t) => t.box.size.xlarge,

            },
        },
    },

};
