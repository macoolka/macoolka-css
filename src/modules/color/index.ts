import { Props, rule } from './rule';
import { theme, Theme, selector } from './theme';
export {
    rule,
    Props,
    theme,
    Theme,
    selector
};
