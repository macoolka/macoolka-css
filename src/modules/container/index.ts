/**
 * Border
 * @prop
 */
import { Props } from '../../basic';
import { Rule } from '../../base/rule';
export type Theme = {
    width: {
        content: number,
        gutter: number;
        nav: number;
        third: number;
    },
    zIndex: {
        moon: number,
        tooltip: number,
        alertDesktop: number,
        popup: number,
        modal: number,
        overlay: number,
        dropdown: number,
        alertMobile: number,
        nav: number,
        bar: number,
        base: number,
    },
};
export const theme: Theme = {
    zIndex: {
        moon: 9999,
        tooltip: 9990,
        alertDesktop: 9000,
        popup: 8700,
        modal: 8500,
        overlay: 8000,
        dropdown: 7000,
        alertMobile: 5000,
        nav: 500,
        bar: 200,
        base: 1,
    },
    width: {
        content: 1160,
        gutter: 40,
        nav: 256,
        third: 355,
    },
};
export type SProps = {
    /**
     * zIndex
     */
    mkZIndex: 'moon' | 'tooltip' | 'alertDesktop' | 'popup' | 'modal' | 'overlay'
    | 'dropdown' | 'alertMobile' | 'nav' | 'bar' | 'base',
};
type EProps = {
    /**
     * scroll bar
     */
    mkScrollBar: 'horizontal' | 'vertical' | 'both' | 'none';

    /**
     * visibility
     */
    mkVisibility: 'hidden' | 'none' | 'elementInvisible' | 'hiddenWidth' | 'hiddenHeight',
    /**
     * align
     */
    mkAlign: 'imageCenter' | 'center',
    /**
     * layout
     */
    mkLayout: 'center' | 'column' | 'row' | 'inlineColumn' | 'inlineRow' | 'inlineCenter',
    /**
     * position
     */
    mkPosition: 'fixedLeftTop' | 'fixedRightTop' | 'fixedTop',
};
export type Props = SProps & EProps;
export const rule: Rule<SProps, EProps, Props, Theme> = {
    rule: {
        mkZIndex: (value, t) => ({ zIndex: t.zIndex[value] }),
    },
    ruleEnum: {
        mkScrollBar: {
            horizontal: {
                overflowX: 'auto',
                overflowY: 'hidden',
            },
            vertical: {
                overflowX: 'hidden',
                overflowY: 'auto',
            },
            both: {
                overflowX: 'auto',
                overflowY: 'auto',
            },
            none: {
                overflowX: 'hidden',
                overflowY: 'hidden',
            },
        },

        mkVisibility: {
            hidden: {
                visibility: 'hidden',
            },
            none: {
                display: 'none',
                visibility: 'hidden',
            },
            elementInvisible: {
                position: 'fixed',
                opacity: 0,
                pointerEvents: 'none',
                margin: '0',
                padding: '0',
                width: '0',
                height: '0',
            },
            hiddenWidth: {
                width: 0,
            },
            hiddenHeight: {
                height: 0,
            },
        },
        mkAlign: {
            imageCenter: {
                backgroundPosition: 'center',
                backgroundRepeat: 'no-repeat',
            },
            center: {
                margin: 'auto',

            },
        },
        mkLayout: {
            center: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
            },
            column: {
                display: 'flex',
                flexDirection: 'column',
            },
            row: {
                display: 'flex',
            },
            inlineCenter: {
                display: 'inline-flex',
                alignItems: 'center',
                justifyContent: 'center',
            },
            inlineRow: {
                display: 'inline-flex',
            },
            inlineColumn: {
                display: 'inline-flex',
                flexDirection: 'column',
            },
        },
        mkPosition: {
            fixedLeftTop: {
                position: 'fixed',
                left: '0',
                top: '0',

            },
            fixedRightTop: {
                position: 'fixed',
                top: '0',
                right: '0',

            },
            fixedTop: {
                position: 'fixed',
                left: '0',
                top: '0',
                right: '0',

            },
        },
    },
};
