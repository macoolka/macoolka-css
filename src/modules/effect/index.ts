
/**
 * Border
 * @prop
 */
import { UnitProps } from '../../basic';
import { Rule } from '../../base/rule';
import { isString } from 'mocoolka-fp/lib/predicate';
import { fromPredicate } from 'mocoolka-fp/lib/Either';
import { Animations } from './animations';
const selectAnimation = (a: Animations) => (t: Theme) => t.effect.animations[a] ? t.effect.animations[a] : a;
export type Theme = {
    effect: {
        animations: { [k in Animations]?: string },
        ease: {
            easeInOut: string,
            easeOut: string,
            easeIn: string,
            sharp: string,
        },
        duration: {
            none: number,
            shortest: number,
            shorter: number,
            short: number,
            standard: number,
            complex: number,
            enteringScreen: number,
            leavingScreen: number,
            long: number,
            longer: number,
            longest: number,
        },
        shadows: {
            0: string,
            1: string,
            2: string,
            3: string,
            4: string,
            5: string,
            6: string,
            7: string,
            8: string,
            9: string,
            10: string,
            11: string,
            12: string,
            13: string,
            14: string,
            15: string,
            16: string,
            17: string,
            18: string,
            19: string,
            20: string,
            21: string,
            22: string,
            23: string,
            24: string,

        },
    }
};
export const theme: Theme = {
    effect: {
        animations: {},
        ease: {
            easeInOut: 'cubic-bezier(0.4, 0, 0.2, 1)',
            easeOut: 'cubic-bezier(0.0, 0, 0.2, 1)',
            easeIn: 'cubic-bezier(0.4, 0, 1, 1)',
            sharp: 'cubic-bezier(0.4, 0, 0.6, 1)',
        },
        duration: {
            none: 0,
            shortest: 150,
            shorter: 200,
            short: 250,
            standard: 300,
            complex: 375,
            enteringScreen: 225,
            leavingScreen: 195,
            long: 450,
            longer: 650,
            longest: 1000,
        },
        // tslint:disable
        shadows: {
            0: 'none',
            1: '0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)',
            2: '0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)',
            3: '0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)',
            4: '0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)',
            5: '0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)',
            6: '0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)',
            7: '0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)',
            8: '0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)',
            9: '0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)',
            10: '0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)',
            11: '0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)',
            12: '0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)',
            13: '0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)',
            14: '0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)',
            15: '0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)',
            16: '0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)',
            17: '0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)',
            18: '0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)',
            19: '0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)',
            20: '0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)',
            21: '0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)',
            22: '0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)',
            23: '0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)',
            24: '0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)',
        },
        //tslint:enabled
    },
}

export type SProps = {

    /**
     * rotate
     */
    mkRotate: number,
    mkTransition: string | {
        property: string,
        duration?: keyof Theme['effect']['duration'],
        ease?: keyof Theme['effect']['ease'],
        delay?: keyof Theme['effect']['duration'],
    };
    mkAnimation: string | {
        name: Animations,
        duration?: keyof Theme['effect']['duration'],
        ease?: keyof Theme['effect']['ease'],
        delay?: keyof Theme['effect']['duration'],
        count?: number | 'infinite',
    };
    mkShadow: keyof Theme['effect']['shadows'],
};
export type EProps = {
    /**
     * flip
     */
    mkFlip: 'horizontal' | 'vertical',


};
export type Props = EProps & SProps;
export const rule: Rule<SProps, EProps, UnitProps, Theme> = {
    ruleEnum: {
        mkFlip: {
            horizontal: a=>a?({
                transform: 'scale(-1, 1)',
            }):{},
            vertical: a=>a?({
                transform: 'scale(1, -1)',
            }):{},
        }
    },
    rule: {
        mkRotate: (a: number) => ({ transform: `rotate(${a}deg)` }),
        mkTransition: (a, t) => fromPredicate(isString, c => c)(a).fold(({
            property,
            duration='shorter',
            ease='easeInOut',
            delay='none',
        }: {
            property: string,
            duration: keyof Theme['effect']['duration'],
            ease: keyof Theme['effect']['ease'],
            delay: keyof Theme['effect']['duration'],
        }) => ({
            transitionProperty: `${property}`,
            transitionDuration: t.effect.duration[duration],
            transitionTimingFunction: `${t.effect.ease[ease]}`,
            transitionDelay: t.effect.duration[delay],
        }), (r: string) => ({
            transitionProperty: r,
            transitionDuration: t.effect.duration.shorter,
            transitionTimingFunction: t.effect.ease.easeInOut,
            transitionDelay: t.effect.duration.none,
        })),
        mkAnimation: (a, t) => fromPredicate(isString, c => c)(a).fold(({
            name,
            duration='shorter',
            ease='easeInOut',
            delay='none',
            count=1,
        }: {
            name: Animations,
            duration: keyof Theme['effect']['duration'],
            ease: keyof Theme['effect']['ease'],
            delay: keyof Theme['effect']['duration'],
            count: number | 'infinite'
        }) => ({
            animationName: selectAnimation(name)(t),
            animationDuration: t.effect.duration[duration],
            animationTimingFunction: t.effect.ease[ease],
            animationDelay: t.effect.duration[delay],
            animationIterationCount: count
        }), (r: Animations) => ({
            animationName: selectAnimation(r)(t),
            animationDuration: t.effect.duration.shorter,
            animationTimingFunction: t.effect.ease.easeInOut,
            animationDelay: t.effect.duration.none,
            animationIterationCount: 1
        })),
        mkShadow: (a, t) => ({ boxShadow: t.effect.shadows[a] })
    }
};

export default rule;
