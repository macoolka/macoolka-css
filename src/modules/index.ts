import * as border from './border';
import * as box from './box';
import * as color from './color';
import * as container from './container';
import * as effect from './effect';
import * as flex from './flex';
import * as text from './text';
import { CssNode } from '../base/CssNode';
import {Rule, Input as RuleInput, parse as parseRule, parseRule as _parseRule } from '../base/rule';
import * as media from '../basic/media';
import {parseMedia, parseUnit, MediaInput, UnitOutput} from '../basic';
// import * as unit from '../basic/unit';
import { BaseProps } from '../base/types';
import { compose } from 'mocoolka-fp/lib/function';
export { toString } from '../base/print';
import { concatRule } from '../base/rule';
export type Theme = border.Theme & box.Theme & color.Theme & container.Theme & effect.Theme & text.Theme & media.Theme;
export const theme: Theme = { ...border.theme, ...box.theme, ...color.theme,
     ...container.theme, ...effect.theme, ...text.theme , ...media.theme};
const rule = concatRule(concatRule(concatRule(
    concatRule(concatRule(concatRule(border.rule, box.rule), color.rule), container.rule)
    , effect.rule), flex.rule), text.rule);
type _Props = border.Props & box.Props & color.Props& container.Props& effect.Props& flex.Props& text.Props;
export type Props= MediaInput<UnitOutput<_Props>>;
export type ModuleOutput<P>= RuleInput<P, Props>;
export const parseModule = <P, T= any>(t: T & Theme) => (i: CssNode<ModuleOutput<P>>): CssNode<BaseProps> =>
compose(parseUnit(t), parseRule(rule)(t), parseMedia<UnitOutput<_Props>>(t))(i);
/* export const parsePropRule = <I, IEnum extends { [key: string]: string }, T= any,
    >(rules: RuleC<I, IEnum, Props, T>) => (t: T & Theme) =>
        (i: CssNode<MediaInput<ModuleOutput<I&IEnum>>>): CssNode<BaseProps> => {
               const mediaResult= parseMedia<ModuleOutput<I&IEnum>>(t)(i);
              return  parseRuleC(parseModule(t))(rules as any)(t)(mediaResult);

        }; */
export const parseModuleRule = <I, IEnum extends { [key: string]: string }, T= any,
        >(rules: Rule<I, IEnum, Props, T>) => (t: T&Theme) =>
            (i: CssNode<ModuleOutput<I&IEnum>>): CssNode<BaseProps> =>
                  _parseRule(parseModule(t))(rules)(t)(i);

/* export const parse = (t: Theme) =>
    (i: CssNode<RuleInput<media.Props<Props>, Props>>): CssNode<BaseProps> => {
        const meidaMap = ruleMap<media.Props<Props>, {}, Props, Theme>(media.rule<Props>())(t);
        const unitRule = ruleMap<unit.Props, {}, BaseProps, Theme>(unit.rule)(t);
        return map(i, compose(unitRule, mmap(theme), meidaMap));
    }; */
