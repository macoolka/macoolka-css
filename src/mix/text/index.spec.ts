import { parseModuleProp as _parse,theme} from '../../modules';
import {rule} from './index'
const parse = _parse(rule,theme);

describe('text', () => {
    it('parse mkTypography', () => {
        expect(parse({
            mkTypography: 'h1',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'h2',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'h3',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'h4',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'h5',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'h6',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'subtitle1',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'subtitle2',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'body1',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'body2',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'button',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'caption',
        })).toMatchSnapshot();
        expect(parse({
            mkTypography: 'overline',
        })).toMatchSnapshot();
    });
   

});

