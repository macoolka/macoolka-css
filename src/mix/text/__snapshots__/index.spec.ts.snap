// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text parse mkTypography 1`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "96px",
  "fontWeight": 300,
  "letterSpacing": "-.01562em",
  "lineHeight": "6rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 2`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "60px",
  "fontWeight": 300,
  "letterSpacing": "-.00833em",
  "lineHeight": "3.75rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 3`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "48px",
  "fontWeight": 400,
  "letterSpacing": "0em",
  "lineHeight": "3.125rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 4`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "32px",
  "fontWeight": 400,
  "letterSpacing": ".00735em",
  "lineHeight": "2.5rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 5`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "24px",
  "fontWeight": 400,
  "letterSpacing": "0em",
  "lineHeight": "2rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 6`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "20px",
  "fontWeight": 500,
  "letterSpacing": ".0125em",
  "lineHeight": "2rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 7`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "16px",
  "fontWeight": 400,
  "letterSpacing": ".00937em",
  "lineHeight": "1.75rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 8`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "14px",
  "fontWeight": 500,
  "letterSpacing": ".00714em",
  "lineHeight": "1.375rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 9`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "16px",
  "fontWeight": 500,
  "letterSpacing": ".03125em",
  "lineHeight": "1.5rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 10`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "14px",
  "fontWeight": 400,
  "letterSpacing": ".01786em",
  "lineHeight": "1.25rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 11`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "14px",
  "fontWeight": 500,
  "letterSpacing": ".08929em",
  "lineHeight": "2.25rem",
  "textDecoration": "none",
  "textTransform": "uppercase",
}
`;

exports[`text parse mkTypography 12`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "12px",
  "fontWeight": 400,
  "letterSpacing": ".03333em",
  "lineHeight": "1.25rem",
  "textDecoration": "inherit",
  "textTransform": "inherit",
}
`;

exports[`text parse mkTypography 13`] = `
Object {
  "fontFamily": "Roboto,Open Sans, Helvetica Neue, Helvetica,Arial, sans-serif",
  "fontSize": "10px",
  "fontWeight": 500,
  "letterSpacing": ".16667em",
  "lineHeight": "2rem",
  "textDecoration": "inherit",
  "textTransform": "uppercase",
}
`;
